test/distributed/org/apache/cassandra/distributed/test/SSTableGenerationTest.java

```diff
@@ -0,0 +1,177 @@
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.distributed.test;
+
+import java.io.IOException;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import org.apache.cassandra.cql3.UntypedResultSet;
+import org.apache.cassandra.db.Directories;
+import org.apache.cassandra.db.Keyspace;
+import org.apache.cassandra.db.SystemKeyspace;
+import org.apache.cassandra.distributed.Cluster;
+import org.apache.cassandra.distributed.shared.ClusterUtils;
+import org.apache.cassandra.io.sstable.Descriptor;
+import org.apache.cassandra.io.sstable.SequenceBasedSSTableUniqueIdentifier;
+import org.apache.cassandra.io.sstable.UUIDBasedSSTableUniqueIdentifier;
+import org.apache.cassandra.metrics.RestorableMeter;
+import org.apache.cassandra.tools.SystemExitException;
+import org.apache.cassandra.utils.UUIDGen;
+import org.assertj.core.api.Assertions;
+import org.assertj.core.data.Offset;
+
+import static org.apache.cassandra.cql3.QueryProcessor.executeInternal;
+import static org.apache.cassandra.db.SystemKeyspace.LEGACY_SSTABLE_ACTIVITY;
+import static org.apache.cassandra.db.SystemKeyspace.SSTABLE_ACTIVITY_V2;
+import static org.apache.cassandra.distributed.shared.FutureUtils.waitOn;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SSTableGenerationTest extends TestBaseImpl
+{
+    private final static String ENABLE_UUID_FIELD_NAME = "enable_uuid_generation_identifiers";
+
+    private volatile int cnt;
+
+    @BeforeClass
+    public static void beforeClass() throws Throwable
+    {
+        TestBaseImpl.beforeClass();
+        ClusterUtils.preventSystemExit();
+    }
+
+    private void createSSTable(Cluster cluster)
+    {
+        cluster.get(1).executeInternal("INSERT INTO " + KEYSPACE + ".tbl (pk, v) values (?,?)", cnt++, cnt++);
+        cluster.get(1).flush(KEYSPACE);
+    }
+
+    private void assertSSTablesCount(Cluster cluster, int expectedSeqGenIds, int expectedUUIDGenIds)
+    {
+        cluster.get(1).runOnInstance(() -> {
+            Set<Descriptor> descs = Keyspace.open(KEYSPACE)
+                                            .getColumnFamilyStore("tbl")
+                                            .getDirectories()
+                                            .sstableLister(Directories.OnTxnErr.THROW)
+                                            .list()
+                                            .keySet();
+            assertThat(descs.stream().filter(desc -> desc.generation instanceof SequenceBasedSSTableUniqueIdentifier).count()).isEqualTo(expectedSeqGenIds);
+            assertThat(descs.stream().filter(desc -> desc.generation instanceof UUIDBasedSSTableUniqueIdentifier).count()).isEqualTo(expectedUUIDGenIds);
+        });
+    }
+
+    @Test
+    public void testRestartWithUUIDEnabled() throws IOException, ExecutionException, InterruptedException
```

Comment:
`ExecutionException` and `InterruptedException` are not thrown. That leaves the import for `ExecutionException` unused.